internal/frontend/selfservice.go

```diff
@@ -78,37 +133,56 @@ func useSelfServiceForm(n core.Nexus) HandlerStep {
 					Value: userEMailAddressSnippet.Render(user),
 				},
 				h.SelectFieldSpec{
-					Name:     "memberships",
+					Name:     Memberships,
 					Label:    "Group memberships",
 					Options:  memberships,
 					ReadOnly: true,
 				},
 				h.MultilineInputFieldSpec{
-					Name:  "ssh_public_keys",
+					Name:  sshPublicKeys,
 					Label: "SSH public key(s)",
 				},
 				h.FieldSet{
-					Name:       "change_password",
+					Name:       changePassword,
 					Label:      "Change password",
 					IsFoldable: true,
 					Fields: []h.FormField{
 						h.InputFieldSpec{
 							InputType: "password",
-							Name:      "old_password",
+							Name:      oldPassword,
 							Label:     "Old password",
 						},
 						h.InputFieldSpec{
 							InputType: "password",
-							Name:      "new_password",
+							Name:      newPassword,
 							Label:     "New password",
 						},
 						h.InputFieldSpec{
 							InputType: "password",
-							Name:      "repeat_password",
+							Name:      repeatPassword,
 							Label:     "Repeat password",
 						},
 					},
 				},
+				h.FieldSet{
+					Name:       configureTotp,
+					Label:      "Configure TOTP",
+					IsFoldable: true,
+					Fields: []h.FormField{
+						// TODO: change to QRCode saved into qrCode
+						h.StaticField{
+							Label: "Secret",
+							Value: codeTagSnippet.Render(key.URL()),
```

Comment:
We could avoid already presenting the totp code here by moving it to a new page. This would also avoid generating the secret everytime someone visits the selfservice page that has not enabled totp.
We can't re-use the code in that case because then someone could just copy the secret from a e.g. screenshare before totp is enabled and retain it as a 
 page refresh wouldn't generate a new secret.

Of course we could use JS to fetch it on demand.

Reply:
Moving to a separate page sounds good. I don't have any JS on the frontend right now, and I intend to wear that as a badge of honor until absolutely necessary.